Editor/WeaverKit/Pipelines/WeaveAssemblies.cs

```diff
@@ -0,0 +1,216 @@
+            public string[] excludePlatforms;
+            public string[] precompiledReferences;
+            public string[] defineConstraints;
+        }
+
+        [PathReferenceResolver, Tooltip("Location where built assemblies will be cached before being staged")]
+        public string assemblyArtifactPath = "<AssemblyStaging>";
+        [HideInInspector]
+        public string UNetAssembly = string.Empty;
+
+        public static string WeaverDir = "<WeaverDir>";
+        public static string WeaverExe = "<WeaverExe>";
+        public static string WeaverTemp = "<WeaverTemp>";
+
+        public override Task Execute(Pipeline pipeline)
+        {
+            var resolvedArtifactPath = PathReference.ResolvePath(assemblyArtifactPath, pipeline, this);
+
+            var definitionDatums = pipeline.Manifest.Data.OfType<AssemblyDefinitions>().ToArray();
+            if (!definitionDatums.Any())
+            {
+                var scriptPath = UnityWebRequest.EscapeURL(AssetDatabase.GetAssetPath(MonoScript.FromScriptableObject(this)));
+                pipeline.Log(LogLevel.Warning, $"No AssemblyDefinitions found, skipping.");
+                return Task.CompletedTask;
+            }
+
+            for (int i = 0; i < definitionDatums.Length; i++)
+            {
+                var datum = definitionDatums[i];
+                if (!datum)
+                    continue;
+
+                var hasUnassignedDefinition = datum.definitions.Any(def => !(bool)def);
+                if (hasUnassignedDefinition)
+                    pipeline.Log(LogLevel.Warning, $"AssemblyDefinitions with unassigned definition at index {i}");
+            }
+
+            var deserializedAsmDefs = definitionDatums.SelectMany(datum =>
+            datum.definitions.Where(asmDefAsset => asmDefAsset)
+            .Select(asmDefAsset => (asmDef: JsonUtility.FromJson<AsmDef>(asmDefAsset.text),
+            datum: datum)));
+            try
+            {
+                string unityEngineArg = GetCoreDLL();
+                string uNetDLLArg = GetUNetAssemblyPath();
+                string outputArg = Path.GetFullPath(resolvedArtifactPath);
+                string extraAssemblyArg = GetExtraAssemblies(pipeline);
+                string[] assemblyToWeaveArgs = GetPathOfAssemblyToWeave(deserializedAsmDefs, resolvedArtifactPath);
+
+                foreach (string assemblyPath in assemblyToWeaveArgs)
+                {
+                    List<string> args = new List<string> { unityEngineArg, uNetDLLArg, outputArg, assemblyPath, extraAssemblyArg };
+                    WeaveAssembly(args, pipeline);
+                }
+            }
+            finally
+            {
+                string tempPath = PathReference.ResolvePath(WeaverTemp, pipeline, this);
+                if (Directory.Exists(Path.GetFullPath(tempPath)))
+                {
+                    Directory.Delete(Path.GetFullPath(tempPath), true);
+                }
+            }
+
```

Comment:
Here, use await Task.WaitAll against the Collection of tasks created above

Reply:
i've set it to return Task.WaitAll(taskCollection), i reckon that's all that i should do for that part, doesnt seem too difficult.

Reply:
specifically it looks like this, since Waitall returns void, i cant use Return Task.WaitAll()

            Task.WaitAll(tasks.ToArray());
            return Task.CompletedTask;